File: spring/src/main/java/com/hcsy/spring/po/SubCategory.java
package com.hcsy.spring.po;

import java.time.LocalDateTime;

import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

import lombok.Data;

@Data
@TableName("sub_category") // 对应数据库中的表名
public class SubCategory {
    @TableId
    private Long id;
    private String name;
    private Long categoryId; // 关联的主分类ID
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

File: fastapi/fastapi-ai/entity/po/subCategory.py
from sqlmodel import SQLModel, Field
from typing import Optional
from datetime import datetime

class SubCategory(SQLModel, table=True):
    __tablename__ = "sub_category"
    id: Optional[int] = Field(default=None, primary_key=True, index=True)
    name: str
    category_id: Optional[int] = None  # 所属分类ID
    create_time: Optional[datetime] = None
    update_time: Optional[datetime] = None